### PR TITLE
Remove redundant PubSummary.lockedVersion

### PR DESCRIPTION
**Pull request by kevmoo**
But keep the logic around to validate the pub output parsing

### CODE REVIEW

**Comment by kevmoo** [comment]
> Can you confirm that if there is an assert issue triggered, it will end up in AnalyzerIssue?

Done. PTAL at the new code...

File: lib/pana.dart [hunk 1]
```diff
@@ -79,8 +79,18 @@ class PackageAnalyzer {
 
     PubSummary summary;
     if (upgrade.exitCode == 0) {
-      summary = PubSummary.create(upgrade.stdout, path: pkgDir);
-      log.info("Package version: ${summary.pkgVersion}");
+      try {
+        summary = PubSummary.create(upgrade.stdout, path: pkgDir);
+      } catch (e, stack) {
+        log.severe("Problem summarizing package", e, stack);
+        //(TODO)kevmoo - should add a helper that handles logging exceptions
```
Comment (kevmoo) [inline lib/pana.dart]: @isoos sound good to you? Just add an inline function?

File: lib/src/pub_summary.dart [hunk 2]
```diff
@@ -13,15 +13,13 @@ import 'utils.dart';
 
 class PubSummary {
   static final _solvePkgLine = new RegExp(
-      r"(?:[><\+ ]) (\w+) (\S+)(?: \((\S+) available\))?(?: from .+)?");
+      r"(?:[><\+\! ]) (\w+) (\S+)(?: \((\S+) available\))?(?: from .+)?");
```
Comment (kevmoo) [inline lib/src/pub_summary.dart]: Oops! Didn't handle the override case. Doing the validation actually helped me fix a bug

File: lib/pana.dart (hunk 1 above)
Comment (isoos) [inline lib/pana.dart]: I don't mind writing two lines of code in such cases, but I don't oppose the helper either.

**Comment by isoos** [comment]
> Done. PTAL at the new code...

LGTM

**Pull request merged**